src/server/scripts/EasternKingdoms/zone_tol_barad.cpp

```diff
@@ -0,0 +1,135 @@
+/*
+ * Copyright (C) 2011-2021 Project SkyFire <https://www.projectskyfire.org/>
+ * Copyright (C) 2008-2021 TrinityCore <http://www.trinitycore.org/>
+ * Copyright (C) 2005-2021 MaNGOS <https://www.getmangos.eu/>
+ * Copyright (C) 2006-2014 ScriptDev2 <https://github.com/scriptdev2/scriptdev2/>
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by the
+ * Free Software Foundation; either version 3 of the License, or (at your
+ * option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for
+ * more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+
+#include "BattlefieldMgr.h"
+#include "BattlefieldTB.h"
+#include "Battlefield.h"
+#include "ScriptSystem.h"
+#include "WorldSession.h"
+#include "ScriptedCreature.h"
+#include "ScriptedGossip.h"
+#include "SpellScript.h"
+#include "Player.h"
+
+enum TBSpiritGuide
+{
+    SPELL_CHANNEL_SPIRIT_HEAL = 22011,
+
+    GOSSIP_OPTION_ID_SLAGWORKS = 0,
+    GOSSIP_OPTION_ID_IRONCLAD_GARRISON = 1,
+    GOSSIP_OPTION_ID_WARDENS_VIGIL = 2,
+    GOSSIP_OPTION_ID_EAST_SPIRE = 3,
+    GOSSIP_OPTION_ID_WEST_SPIRE = 4,
+    GOSSIP_OPTION_ID_SOUTH_SPIRE = 5,
+};
+
+class npc_tb_spirit_guide : public CreatureScript
```

Comment:
duno about this whole script you should not be able to choose where you want to go. you should be ported away from a capped gy when the enemy faction gets the cap to a friendly gy